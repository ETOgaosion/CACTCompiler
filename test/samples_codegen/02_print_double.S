iobuff:
	.zero 80
	.equ	ImmValue_0, 0x400b333333333333

.text
.align 1
.globl	main
.type	main, @function
main:
	sd	ra, -8(sp)
	li	t0, ImmValue_0
	sd	t0, -16(sp)
	fld	fa0, -16(sp)
	addi	sp, sp, -16
	call	print_double
	addi	sp, sp, 16
	mv	a0, zero
	ld	ra, -8(sp)
	ret
