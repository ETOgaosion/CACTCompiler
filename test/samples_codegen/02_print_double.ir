============ global var ================
============ imm value ================
ImmValue_0: 3.4
============ functions ================
----------- funcDef -----------
main() => 2:
----------- paramVars -----------
----------- localVars -----------
a := symbol type: 1; data type: 4; initValue: ImmValue_0: 3.4

----------- tempVars ------------
----------- Codes ------------
main:
	a = ImmValue_0: 3.4;
	addparam a;
	call print_double; 
	return 0;

