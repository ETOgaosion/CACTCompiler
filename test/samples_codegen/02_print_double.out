bbl loader
3.400000
